enhancements/insights/pulling-pull-secret-from-ocm.md

```diff
@@ -0,0 +1,191 @@
+4. The Insights operator calls this API periodically, fetches the updated `pull-secret` and compares it with the current global `pull-secret` on the cluster (only entries available in both pull-secrets are compared).
+5. If there is a mismatch, the Insights operator applies the new `pull-secret`.
+   * The updated `pull-secret` from the OCM API is an authoritative source. Insights operator replaces and adds (There should not be any new entries, but this may change in the future) the entries provided in the updated `pull-secret` from the OCM API (This means that the Insights operator cannot delete anything in the existing secret).
+   * Note: MCO updating the `pull-secret` on the nodes will not result in node reboots. This will require an update of the documentation in https://docs.openshift.com/container-platform/4.9/updating/update-using-custom-machine-config-pools.html#update-using-custom-machine-config-pools-about_update-using-custom-machine-config-pools.
+   We should emphasize the need of unpausing the machine config pools to succesfully apply the new `pull-secret` (credentials) to all the nodes.
+6. The Insights operator itself starts using the new `pull-secret`
+7. Cluster comes up with the new `pull-secret` and starts reporting back to OCM into the new account.
+
+### User Stories
+
+As an OCM user, I would like to transfer cluster ownership from my account to another user's account in one step.
+- [CCXDEV-6529 Insights Operator pulling updated pull-secret to transfer unclassified cluster](https://issues.redhat.com/browse/CCXDEV-6529)
+- [CCX-185 Transfer of Unclassified clusters - provide pull-secret](https://issues.redhat.com/browse/CCX-185)
+### API Extensions
+
+No API extensions.
+
+### Risks and Mitigations
+
+Risk: The OCM API is down or returning HTTP response code >= 500.
+
+Mitigation: The Insights Operator is marked as degraded in such case.
+
+Risk: The `pull-secret` does not work or is otherwise corrupted.
+
+Mitigation: We must refer to the OCP documentation, which describes the manual fix procedure.
+
+Risk: Users don't want to wait 24 hours for the pull secret to be updated on the cluster
+
+Mitigation: User can trigger the update faster by deleting the Insights operator Pod (This needs to be documented and may change in the future - e.g we can introduce a new CRD to configure this).
+
+## Design Details
+
+### Authorization
+
+The Insights Operator is able to pull the updated `pull-secret` content from the OCM API using the existing `cloud.openshift.com` token
+available in the `pull-secret` (in the `openshift-config` namespace).
+
+The Insights Operator must provide a cluster ID as an identifier of the cluster.
+
+### OCM API
+
+The OCM API provides an endpoint that will serve the updated `pull-secret` definition when the migration window is opened. The endpoint will be defined in a new `ClusterTransfer` API - more details are in https://gitlab.cee.redhat.com/service/uhc-account-manager/-/merge_requests/2716.
```

Comment:
you can change this link from MR to the doc itself:  https://gitlab.cee.redhat.com/service/uhc-account-manager/-/blob/master/docs/cluster_transfer_api.md

Reply:
Updated. Thanks